src/Microsoft.DocAsCode.Build.Common/YamlHeaderParser.cs

```diff
@@ -7,57 +7,60 @@ namespace Microsoft.DocAsCode.Build.Common
     using System.Collections.Generic;
     using System.IO;
     using System.Linq;
-    using System.Text.RegularExpressions;
 
     using Microsoft.DocAsCode.Common;
     using Microsoft.DocAsCode.DataContracts.Common;
+    using Microsoft.DocAsCode.MarkdownLite;
 
-    /// <summary>
-    /// Match yaml header from markdown files.
-    /// YAML HEADER Syntax
-    /// 1. Started with a new line
-    /// 2. Followed by three dashes `---` as a line, spaces are allowed before 
-    /// 3. Followed by and must followed by `uid: `
-    /// 4. Followed by other properties in YAML format
-    /// 5. Ended with three dashes `---` and environment newline as a line, spaces are allowed before 
-    /// </summary>
     public static class YamlHeaderParser
     {
-        private static readonly List<string> RequiredProperties = new List<string> { Constants.PropertyName.Uid };
-
-        // If is not the end of the file, then \n should be appended to ---
-        public static readonly Regex YamlHeaderRegex = new Regex(@"((((?!\n)\s)*\n)|^)((?!\n)\s)*\-\-\-((?!\n)\s)*\n((?!\n)\s)*(?<content>uid:.*?)\s*\-\-\-((?!\n)\s)*\n", RegexOptions.Compiled | RegexOptions.Singleline);
-
-        public static IList<MatchDetail> Select(string input)
+        public static IEnumerable<MatchDetail> Select(string html)
         {
-            if (string.IsNullOrEmpty(input)) return null;
-            var yamlHeader = YamlHeaderRegex.Matches(input);
-            return (from Match item in yamlHeader select SelectSingle(item, input)).ToList();
+            if (string.IsNullOrEmpty(html)) return null;
+            var parts = YamlHtmlPart.SplitYamlHtml(html);
+
+            var details = parts.Select(SelectSingle).Where(o => o != null);
+            return details != null ? details : Enumerable.Empty<MatchDetail>();
```

Comment:
sure

---
In reply to: [65115821](https://github.com/dotnet/docfx/pull/343#discussion_r65115821) [](ancestors = 65115821)